### PR TITLE
docs(adrs): align implementation status with codebase audit

### PR DESCRIPTION
**Pull request by djdanielsson**
## Summary

- Audited 21 ADRs previously listed as "Accepted but not yet fully implemented" against the codebase
- Moved **10 ADRs to `Implemented`** where the decision is fully reflected in shipped code
- Introduced **`Partially Implemented`** as a first-class ADR status for substantial but incomplete delivery
- Migrated partial ADRs from overloaded `Accepted (...)` lines to `Partially Implemented`
- Fixed `scripts/generate_adr_index.py` status normalization
- Added **`/pr-traceability`** skill and wired it into PR create/review workflows

## Changes

**New lifecycle:** `Proposed → Accepted → Partially Implemented → Implemented`

| Index bucket | Count | Meaning |
|--------------|------:|---------|
| Implemented | 46 | Complete per ADR scope |
| Partially Implemented | 15 | Code shipped; documented gaps remain |
| Accepted | 1 | Decided, not started (ADR-049) |
| Proposed | 7 | Not yet accepted |
| Superseded | 2 | Replaced |

**Partially Implemented (15):** ADR-012, 038, 040, 043, 048, 050, 051, 052, 053, 054, 055, 056, 057, 059, 060

**Accepted (not started):** ADR-049 (Gateway embedded in local daemon)

## Related issues

- Refs traceability/process work — no single closing issue

## Related ADRs

- Multiple ADR status/revision updates across `.sdlc/adrs/ADR-*.md` (see diff)
- Index regenerated via `scripts/generate_adr_index.py`

## Review feedback addressed

- **3a52ee28**: Downgraded ADR-040, 048, 056, 057, 059 to Partially Implemented; clarified ADR-038 pending work; fixed ADR-066 revision table; tightened Accepted status boundary in index generator

## Quality of life

- New **`/pr-traceability`** skill: mandatory ADR + GitHub issue checklist for `pr-new`, `pr-address-feedback`, and `pr-contributor-review`
- Updated `.sdlc/templates/adr.md`, `/sdlc-status`, and PR body template with `Related issues` / `Related ADRs` sections
- Added `tests/test_generate_adr_index.py` for status normalization edge cases

## Test plan

- [x] `tox -e lint` passes (including `adr-index` pre-commit hook and skill frontmatter)
- [x] `tox -e unit -- tests/test_generate_adr_index.py` passes

## Summary by CodeRabbit

* **Documentation**
  * Updated architecture decision records to distinguish implemented capabilities from remaining work across deployment, APIs, scanning, integrations, operations, and UI workflows.
  * Clarified scaling, topology, authentication, release, SCM, and live-state status, and updated the ADR index and implementation history.
  * Added the “Partially Implemented” lifecycle stage to ADR guidance and templates.
  * Added PR traceability guidance for linking related issues and ADRs and keeping their status up to date.
* **Chores**
  * Improved ADR status classification and normalization.
* **Tests**
  * Added coverage for status classification, detail text, and ambiguous descriptions.

### CODE REVIEW

File: .sdlc/adrs/ADR-066-ui-workflow-github-release-artifacts.md [hunk 1]
```diff
@@ -151,3 +151,4 @@ enough and the UI package tracks them naturally.
 |------|--------|
 | 2026-07-23 | Initial — GitHub Release tarball publish for `@apme/ui-workflow` |
 | 2026-07-31 | Consolidated into main APME release; removed separate `ui-workflow-v*` tag convention |
+| 2026-09-22 | Agent | Status → Implemented — `ui-workflow-release.yml` and `tox -e ui-workflow-pack` publish `@apme/ui-workflow` with main release |
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply .sdlc/adrs/ADR-066-ui-workflow-github-release-artifacts.md]: Merged the Agent author into the Change cell so the revision-history row matches the two-column Date/Change header (fixes markdownlint MD056). Fixed in 3a52ee28.

File: .sdlc/adrs/ADR-038-public-data-api.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 ## Status
 
-Accepted
+Accepted (pull API, project lookup, and dashboard implemented; webhooks and token auth pending)
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply .sdlc/adrs/ADR-038-public-data-api.md]: Added `GET /api/v1/projects/{id}/health` to the ADR-038 status pending list and revision history. Fixed in 3a52ee28.

File: scripts/generate_adr_index.py [hunk 3]
```diff
@@ -43,6 +43,10 @@ def _normalize_status(raw: str) -> str:
     low = raw.lower().strip()
     if "superseded" in low:
         return "Superseded"
+    if low == "implemented" or low.startswith("implemented "):
+        return "Implemented"
+    if low.startswith("accepted"):
+        return "Accepted"
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply scripts/generate_adr_index.py]: Tightened Accepted matching to exact `accepted`, `accepted ...`, or `accepted (...)` forms and removed the broad `accepted` substring fallback. Added a test that `Acceptedness` does not bucket as Accepted. Fixed in 3a52ee28.

File: .sdlc/adrs/ADR-056-apme-owns-scm-commit-push.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 ## Status
 
-Accepted
+Implemented
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply .sdlc/adrs/ADR-056-apme-owns-scm-commit-push.md]: Downgraded ADR-056 to Partially Implemented — Gateway SCM submit is shipped, but commits still use the caller token owner rather than the ADR's fixed APME author/committer identity; that gap is now documented as pending. Fixed in 3a52ee28.

File: .sdlc/adrs/ADR-040-scan-metadata-enrichment.md [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 ## Status
 
-Accepted
+Implemented
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply .sdlc/adrs/ADR-040-scan-metadata-enrichment.md]: Downgraded ADR-040 to Partially Implemented — project-scoped manifest, dependencies, and SBOM are shipped, but global `GET /api/v1/collections`, `GET /api/v1/python-packages`, and collection-to-project catalog endpoints remain pending. Fixed in 3a52ee28.

File: .sdlc/adrs/ADR-057-per-rule-version-applicability.md [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 ## Status
 
-Accepted
+Implemented
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply .sdlc/adrs/ADR-057-per-rule-version-applicability.md]: Downgraded ADR-057 to Partially Implemented — engine `version_defaults.py`, proto field, and violation injection are shipped, but Gateway reconciliation does not persist or expose `ansible_core_version` in the rule catalog. Fixed in 3a52ee28.

File: .sdlc/adrs/ADR-048-pod-internal-admin-endpoints.md [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 ## Status
 
-Accepted
+Implemented
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply .sdlc/adrs/ADR-048-pod-internal-admin-endpoints.md]: Downgraded ADR-048 to Partially Implemented — admin endpoints exist, but Galaxy Proxy still binds `0.0.0.0` and Podman publishes `hostPort` 8765; localhost-only binding is documented as pending before Implemented. Fixed in 3a52ee28.

File: .sdlc/adrs/ADR-038-public-data-api.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (djdanielsson) [reply .sdlc/adrs/ADR-038-public-data-api.md]: Clarified ADR-038 pending work: collection-route `GET /api/v1/projects?repo_url=...` is not implemented (lookup exists only at `/projects/lookup`). Status line and revision history updated. Fixed in 3a52ee28.

File: .sdlc/adrs/ADR-059-graph-library-extraction.md [hunk 8]
```diff
@@ -2,7 +2,7 @@
 
 ## Status
 
-Accepted
+Implemented
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply .sdlc/adrs/ADR-059-graph-library-extraction.md]: Downgraded ADR-059 to Partially Implemented — graph package is extracted, but `L030_non_builtin_use_graph` and shared helpers still import `apme_engine.engine` at runtime; remaining Engine boundary imports are documented as pending. Fixed in 3a52ee28.

File: .sdlc/adrs/ADR-040-scan-metadata-enrichment.md [hunk 9]
```diff
@@ -2,7 +2,7 @@
 
 ## Status
 
-Accepted
+Partially Implemented (global collection and Python-package catalog endpoints pending)
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply .sdlc/adrs/ADR-040-scan-metadata-enrichment.md]: Added collection-to-project routes to the ADR-040 status parenthetical so the generated README index reflects the full pending scope (global catalog endpoints plus `GET /api/v1/collections/{fqcn}/projects`). Fixed in ce4782a2.

File: .sdlc/adrs/ADR-048-pod-internal-admin-endpoints.md [hunk 10]
```diff
@@ -157,3 +157,5 @@ could invalidate it.
 | Date | Author | Change |
 |------|--------|--------|
 | 2026-04-01 | APME Team | Initial proposal — accepted |
+| 2026-09-22 | Agent | Status → Implemented — pod-internal admin endpoints rely on network isolation as designed; no auth layer required until topology changes |
+| 2026-09-22 | Agent | Status → Partially Implemented — Galaxy Proxy still binds `0.0.0.0` and Podman publishes `hostPort` 8765; localhost-only binding required before Implemented |
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply .sdlc/adrs/ADR-048-pod-internal-admin-endpoints.md:166]: Updated ADR-048 context and decision to describe current Galaxy Proxy exposure (`0.0.0.0:8765`, Podman `hostPort` 8765) and retained localhost-only binding plus host-port removal as prerequisites for Implemented. Fixed in ce4782a2.

**Comment by djdanielsson** [comment]
@coderabbitai full review

File: .agents/skills/pr-new/SKILL.md [hunk 11]
```diff
@@ -658,6 +679,13 @@ gh pr create --repo upstream-owner/repo --title "conventional commit style title
 ## Changes
 - List of notable changes
 
+## Related issues
+- Closes #NNN — <why>
+- Refs #NNN — <partial / context> (omit section if truly none; say so)
+
+## Related ADRs
+- ADR-NNN — <status change or "reviewed, no change"> (omit section if none)
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply .agents/skills/pr-new/SKILL.md]: Updated the Step 8 PR body template to always include both Related issues and Related ADRs sections, using a `None — <why>` entry when none apply (matching pr-traceability Step 5). Fixed in 2291abfb.
